### ISSUE TITLE
Error: The operation could not complete due to unexpected issues

### ISSUE THREAD

**Issue by Luvuyob** [alextselegidis/easyappointments#379]
I have been trying to install EA and have been getting this error message: **The installation could not be completed due to an unexpected issue. Please check the browser's console for more information.
Close**.

When I investigate I found out that it's a 500 (Internal server error). 

Can you please help as I'm stuck and do not know what to do from here.

**Comment by sjelle** [comment]
Hi Luvuyob,
I had the same error and it was related to wrong file permissions. How did you upload the files ? Did you extract them on the server or on your machine ?
I am not an expert, just trying to help out.

**Comment by Luvuyob** [comment]
Hi Sjelle

I uploaded the zip file onto the server and extracted everything while it
was there. after that I configured the database to the config file. Then I
opened the URL, after filling in the required info and clicked on the
installation button it returned that error message.

**Comment by sjelle** [comment]
Just a quick question, did you transfer the files after unzipping to a subdir ? Or did you leave them untouched ?

**Comment by Luvuyob** [comment]
I uploaded them to the sub directory in the zip file.

**Comment by Jannibal** [comment]
Tl;dr
Make sure that the database user that EA uses has correct permissions (reference permission, e.g.)

I just had the same problem myself.
I solved it by turning on the debug mode in config.php.
Then I tried installing and to do the first configuration with the developer console running. Saved the log by right clicking on it (it's nearly useless to read the error in the console).

In the log file I saw that the wordpress user did not have "reference" permissions to the database, so I added that permission to the user and it worked.

**Comment by Luvuyob** [comment]
I tried but still can't manage to do what you suggested. do you mind assisting me over TeamViewer?

**Comment by alextselegidis** [comment]
@Luvuyob Did you manage to install the app? 

It seems to me as well that this is a database permissions things.

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by Luvuyob** [comment]
No I haven't. The thing is I'm not good with these database stuff. I
recently taught myself because I could not find someone to help me with
creating my site.

**Comment by alextselegidis** [comment]
I'll need some more detailed error info in order to understand the nature of the problem. 

For this it would be helpful to take a look at the error logs of your server and post any related info here. 

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by Luvuyob** [comment]
Hi Alex

I have attached a picture I hope it helps

**Comment by Luvuyob** [comment]
The installation could not be completed due to an unexpected issue. Please check the browser's console for more information.

When I check the console it shows this: 

Error Number: 1146

Table 'ejcspruf_EA.ea_settings' doesn't exist

SELECT * FROM (`ea_settings`) WHERE `name` = 'company_name'

Filename: /home/ejcspruf/bookings.misterbooks.co.za/models/settings_model.php

Line Number: 44

**Comment by alextselegidis** [comment]
Please change your BASE_URL parameter in your config.php to `"http://bookings.misterbooks.co.za" (remove trailing slash). 

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by Luvuyob** [comment]
I've changed it but still is showing the same error message.

**Comment by Luvuyob** [comment]
I've managed to install it and now I can't access the backend. it's not
accepting my credentials.

**Comment by alextselegidis** [comment]
I did the installation yesterday for you while testing your setup. 

The credentials should be administrator / administrator. 

Please do change them asap. 

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by Luvuyob** [comment]
Hi Alex, On the final page after confirming the booking this error shows
up: AJAX Error: error

**Comment by alextselegidis** [comment]
Hey @Luvuyob, it seems that your site is currently unavailable. 

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by Luvuyob** [comment]
what do you mean that it's unavailable?

**Comment by alextselegidis** [comment]
Hi @Luvuyob, I tried to visit your site some days ago and it was down for some reason. I tried again today and it works so no problem with that :) 

It seems that for some reason the mappings made by CodeIgniter database library are not working on your server. 

I would recommend you to install the latest release (v1.3 beta) and try this one cause CodeIgniter is also updated since the Easy!Appointments last release.

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by Luvuyob** [comment]
Where can I get the beta 1.3 from?

**Comment by alextselegidis** [comment]
From easyappointments.org 

Here's the download link: 

https://github.com/alextselegidis/easyappointments/releases/download/1.3.0-beta.1/easyappointments_1.3.0_beta.1.zip

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by Luvuyob** [comment]
Thanks, Alex will try installing it now

**Comment by Luvuyob** [comment]
*Error message that shows up after installing v**1.3*

*Warning*: require_once(/home/ejcspruf/bookings.misterbooks.co.za/config.php):
failed to open stream: No such file or directory in
*/home/ejcspruf/bookings.misterbooks.co.za/index.php
<http://bookings.misterbooks.co.za/index.php>* on line *50*

*Fatal error*: require_once(): Failed opening required '/home/ejcspruf/
bookings.misterbooks.co.za/config.php'
(include_path='.:/usr/lib/php:/usr/local/lib/php') in
*/home/ejcspruf/bookings.misterbooks.co.za/index.php
<http://bookings.misterbooks.co.za/index.php>* on line *50*

**Comment by alextselegidis** [comment]
You'll need to copy your root config-sample.php into the config.php file and set the correct info. 

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by haruiti** [comment]
I just uninstalled and installed again from scratch. So I just replace the original files with the modified ones. I believe the problem was in the database.

**Comment by asiboro** [comment]
> I keep getting "Please remove any trailing slashes from your "BASE_URL" setting of the root "config.php" file and try again. though my BASE_URL is clearly without trailing slashes.
> 
> ```
> // ------------------------------------------------------------------------
> // GENERAL SETTINGS
> // ------------------------------------------------------------------------
> 
> const BASE_URL      = 'https://tlece-cmtors.jpn.org';
> const LANGUAGE      = 'english';
> const DEBUG_MODE    = TRUE;
> ```

This was using easyappointments-1.5.0-dev.2. I switched to 1.4.3 using the same config.php dan exact same server, it worked. So this is a bug with version 1.5.0 dev 2
